server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -158,7 +179,10 @@ func (p *FetchServer) FetchBlob(ctx context.Context, req *rapb.FetchBlobRequest)
 	}
 
 	return &rapb.FetchBlobResponse{
-		Status: &statuspb.Status{Code: int32(gcodes.NotFound)},
+		Status: &statuspb.Status{
+			Code:    int32(gcodes.NotFound),
+			Message: status.Message(fetchErr),
+		},
```

Comment:
can you just do Status: fetchErr.GRPCStatus()?

Reply:
or just add a comment here explaining you intentionally always want to return NotFound for reason x

Reply:
Done